src/shapes/path.class.js

```diff
@@ -86,6 +86,18 @@
 
       ctx.beginPath();
 
+      // SCI-2706 - detect a single point ('m', and all 'l' with the same coords)
+      // Safari doesnt draw zero-length line segments
+      // Safari specs: https://html.spec.whatwg.org/multipage/canvas.html#trace-a-path
+      var addZeroLenOffset = true;
+      for (var i = 0; i < this.path.length - 1; i++) {
+        if (this.path[i][1] !== this.path[i + 1][1] ||
+            this.path[i][2] !== this.path[i + 1][2]) {
+          addZeroLenOffset = false;
```

Comment:
This is done at rendering time?
Assuming we can't do this at path creation time because we don't know what will correspond to 1px? But... this is adding the offset to the core path position so... yeah... why can't this be done at path creation time? I guess it wouldn't help existing save files that way...

Reply:
There are a few considerations here about doing it at path creation time: 
1) As you say it wont help with existing save files. This could be detected and corrected by our save-file versioning, but not sure if it's worth adding a processing step for this. 
2) Our current precision is set to round to the nearest int. So as soon as we made the save file, it would strip the .001. We could change our number-of-digits precision but that would add to our save files (not only just the extra precision, but the code that removes colinear points in the middle of a line would find a lot less things to remove - even if we change this to .1 rather than .001). We can also add an entire pixel offset rather than the .001  but it makes an oval rather than a dot. 

I could be convinced to move this if you think those aren't compelling reasons. Also curious what others think.

Reply:
we probably don't want to do that on load since the user might start seeing dots where they didn't see before.   I think it would make sense at path creation time if possible.

Reply:
My concern when seeing this was performance. This will loop through all points in all paths. Does this become a problem at some point when there are a lot of paths?

Reply:
Ok, revised to modify the path at path-creation -  VWolfe and ESilberman discussed and agreed that it's ok if existing dots in save files don't show up in safari, and after reviewing a screenshot of the 'adjusted' dots are fine with the way they look.